src/Microsoft.ML.StandardLearners/Optimizer/DifferentiableFunction.cs

```diff
@@ -101,7 +101,6 @@ private void Eval(object chunkIndexObj)
             VBuffer<Float> tempGrad = default(VBuffer<Float>);
             for (int i = from; i < to; ++i)
             {
-                tempGrad = new VBuffer<Float>(0, 0, tempGrad.Values, tempGrad.Indices);
```

Comment:
I wasn't sure on this one. Normally when you call a function passing in a `VBuffer`, it is assumed that the function will return with the right result into that VBuffer. So "zeroing" it right before calling `_func` seemed unnecessary.

Reply:
Well, that depends. I wonder if this `_func` here is some sort of "accumulating" gradient, so that if you call it multiple times without a refresh it will accumulate the partial gradients for all things. That could explain this pattern.

---
In reply to: [234085400](https://github.com/dotnet/machinelearning/pull/1650#discussion_r234085400) [](ancestors = 234085400)

Reply:
Actually I tried to go hunting for it to see if the usages of this actually bear out, there are no usages of `DifferentiableFunctionAggregator` at all. Not sure what's up with that. This code though is probably one of the oldest parts of the codebase, so maybe it was used years ago and no one noticed when we stopped using it. To be honest I'm not even sure what it is meant to do...

---
In reply to: [234102794](https://github.com/dotnet/machinelearning/pull/1650#discussion_r234102794) [](ancestors = 234102794,234085400)

Reply:
I'll add this line back.  I've opened https://github.com/dotnet/machinelearning/issues/1653 to investigate removing this class since it isn't used.